### PR TITLE
Prefer `assertThrows` to try/fail

### PR DESCRIPTION
**Pull request by basil**
Some test cleanup, mainly eliminating usages of `try/fail` in favor of the more legible alternative offered by JUnit 4.13 in the form of `Assert#assertThrows`.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@StefanSpieker @offa

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: test/src/test/java/jenkins/security/s2m/DefaultFilePathFilterTest.java
Comment (daniel-beck) [inline test/src/test/java/jenkins/security/s2m/DefaultFilePathFilterTest.java:68]: Should `ReverseCallable` be changed accordingly? It declares that it throws `Exception`, so `#call` does too.

File: test/src/test/java/hudson/model/labels/LabelAtomSecurity1986Test.java
Comment (daniel-beck) [inline test/src/test/java/hudson/model/labels/LabelAtomSecurity1986Test.java:176]: Out of scope, but looks like low-hanging fruit for deduplication

File: test/src/test/java/hudson/model/DescriptorTest.java
Comment (daniel-beck) [inline test/src/test/java/hudson/model/DescriptorTest.java:208]: Not sure whether this one is improved by this change, but it's the only one 😄

**Review by daniel-beck — APPROVED** [review]

**Review by offa — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by StefanSpieker — APPROVED** [review]
Great effort. I think you could also get rid of a couple of superfluous throws in the test definition. So something like
`void testFoo throws Exception` would not need the throws anymore after your change.
But that could also be done afterwards.

File: test/src/test/java/jenkins/security/s2m/DefaultFilePathFilterTest.java
Comment (basil) [reply test/src/test/java/jenkins/security/s2m/DefaultFilePathFilterTest.java:68]: > Should `ReverseCallable` be changed accordingly?

This bound can't be tightened, because `#call` could throw either `IOException` or `InterruptedException`.

File: test/src/test/java/hudson/model/labels/LabelAtomSecurity1986Test.java
Comment (basil) [reply test/src/test/java/hudson/model/labels/LabelAtomSecurity1986Test.java:176]: Indeed

**Comment by basil** [comment]
> Great effort. I think you could also get rid of a couple of superfluous throws in the test definition. So something like
> `void testFoo throws Exception` would not need the throws anymore after your change.
> But that could also be done afterwards.

In many cases the `throws Exception` is still needed due to some other method unrelated to the `assertThrows`. Let's focus on unnecessary throws clauses in a bulk change in a separate PR.

**Pull request merged**
